static/app/utils/duration/formatTraceDuration.tsx

```diff
@@ -2,7 +2,7 @@ const format = (v: number, abbrev: string, precision: number) => {
   if (v === 0) {
     return '0' + abbrev;
   }
-  return v.toFixed(precision) + abbrev;
+  return parseFloat(v.toFixed(precision)) + abbrev;
```

Comment:
I'm not sure this is the ideal solution. While it will fix the problem I reported, it will mean that any values which are actually measured to a greater precision might have zeroes truncated. 

For example, what if we have a value that is measured to 0.01-ms precision to be exactly 7.00ms. If we display the value as "7ms", then it is not clear that the value is accurate to 0.01ms. So, we should display such values as "7.00ms."

To fix the problem I identified, I think it would be preferable to change the `precision` value that we pass to this function. When we are rounding to a whole number of milliseconds, we would pass `precision=0`.

Reply:
Yes, that would be technically more correct, but it would be more confusing to the user as it would result in values being shown as 1.00ms for spans, but 1ms for transactions. This then begs the question of why, which I dont think is something that the user should ever care about. 

We should have a single representation here and either do this or fix the underlying root cause, but I dont want to confuse the users here.

Reply:
I disagree with your reasoning here:

> We should have a single representation here

If transactions have 1-ms precision and spans have 0.01-ms precision, I think it is perfectly reasonable that they would have separate representations. They are different types of data.

Yes, users might wonder why the data has different precisions, which is why we should work to quickly fix the underlying root cause, but in the meantime I think it is a bit deceitful for us to conceal the different precisions of the data. I think it is important that users can easily see how precise the data is, and this suggestion would hide that information from them.

To be fair, this is still an improvement from what we had before

Reply:
Cool, sounds good! And yes, I agree: best solution would be to fix the precision of the transaction timing to go to 0.01-ms